File: my-folder/0026-remove-duplicates-from-sorted-array/solution.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        n = len(nums)

        if n == 1:
            return 1

        i = 0
        j = 1

        while j < n:
            if nums[j] != nums[i]:
                i+=1
                nums[i] = nums[j]
            j+=1
        return i+1

File: my-folder/0026-remove-duplicates-from-sorted-array/solution.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {

        int insertIndex = 1;
        for (int i = 1; i < nums.size(); i++){
            if ( nums[i - 1] != nums[i] )
                {
                    nums[insertIndex] = nums[i];
                    insertIndex+=1;
                }
        }
        return insertIndex;
    } 



    // int n = nums.size();
        // int left = 1;
        // int right = 1;

        // while ( right < n ){
        //     while (right < n && nums[right] == nums[right - 1])
        //         right++;
            
        //     if ( right < n )
        //     {
        //         nums[left] = nums[right];
        //         left+=1;
            
        //     } 
        //     right+=1;
            
        // }
        // return left;  
};
